File: src/engine/model/BehaviourConfiguration.cpp
/*
 * BehaviourConfiguration.cpp
 *
 *  Created on: Mar 5, 2014
 *      Author: Stephan Opfer
 */

#include "engine/model/BehaviourConfiguration.h"

namespace alica
{

	BehaviourConfiguration::BehaviourConfiguration()
	{
		this->eventDriven = false;
		this->frequency = 30;
		this->deferring = 0;
		this->behaviour = NULL;

	}

	BehaviourConfiguration::BehaviourConfiguration(long id) :
			BehaviourConfiguration()
	{
		this->id = id;
	}

	BehaviourConfiguration::~BehaviourConfiguration()
	{

	}

	string BehaviourConfiguration::toString()
	{
		stringstream ss;
		ss << "#BehaviourConfiguration: " << this->getName() << " " << this->getId() << endl;
		ss << "\t Behaviour: ";
		if (this->getBehaviour() != NULL)
		{
			ss << name << " " << this->getBehaviour()->getId();
		}
		ss << endl;
		ss << "\t Deferring: " + this->getDeferring() << endl;
		ss << "\t Frequency: " + this->getFrequency() << endl;
		ss << "\t MasterPlan?: " + this->isMasterPlan() << endl;
		ss << "\t Parameters: " + this->getParameters().size() << endl;

		if (this->getParameters().size() != 0)
		{
			for(map<string,string>::const_iterator iter = this->getParameters().begin(); iter != this->getParameters().end(); iter++)
			{
				const string s = iter->first;
				const string val = iter->second;
				ss << "\t" + s << " : " << val << endl;
			}
		}
		ss << endl;
		ss << "#EndBehaviourConfiguration" << endl;

		return ss.str();
	}

//================== Getter and Setter =========================

	int BehaviourConfiguration::getDeferring() const
	{
		return deferring;
	}

	void BehaviourConfiguration::setDeferring(int deferring)
	{
		this->deferring = deferring;
	}

	bool BehaviourConfiguration::isEventDriven() const
	{
		return eventDriven;
	}

	void BehaviourConfiguration::setEventDriven(bool eventDriven)
	{
		this->eventDriven = eventDriven;
	}

	int BehaviourConfiguration::getFrequency() const
	{
		return frequency;
	}

	void BehaviourConfiguration::setFrequency(int frequency)
	{
		this->frequency = frequency;
	}

	const map<string, string>& BehaviourConfiguration::getParameters() const
	{
		return parameters;
	}

	void BehaviourConfiguration::setParameters(const map<string, string>& parameters)
	{
		this->parameters = parameters;
	}

	const Behaviour* BehaviourConfiguration::getBehaviour() const
	{
		return behaviour;
	}

	void BehaviourConfiguration::setBehaviour(const Behaviour* behaviour)
	{
		this->behaviour = behaviour;
	}

} /* namespace Alica */

File: alica_plan_designer/src/de.uni-kassel.vs.cn.planDesigner.alica/src/de/uni_kassel/vs/cn/planDesigner/alica/BehaviourConfiguration.java
/**
 */
package de.uni_kassel.vs.cn.planDesigner.alica;

import org.eclipse.emf.common.util.EMap;

/**
 * <!-- begin-user-doc -->
 * A representation of the model object '<em><b>Behaviour Configuration</b></em>'.
 * <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * <ul>
 *   <li>{@link de.uni_kassel.vs.cn.planDesigner.alica.BehaviourConfiguration#getParameters <em>Parameters</em>}</li>
 *   <li>{@link de.uni_kassel.vs.cn.planDesigner.alica.BehaviourConfiguration#getDeferring <em>Deferring</em>}</li>
 *   <li>{@link de.uni_kassel.vs.cn.planDesigner.alica.BehaviourConfiguration#getFrequency <em>Frequency</em>}</li>
 *   <li>{@link de.uni_kassel.vs.cn.planDesigner.alica.BehaviourConfiguration#getBehaviour <em>Behaviour</em>}</li>
 *   <li>{@link de.uni_kassel.vs.cn.planDesigner.alica.BehaviourConfiguration#isEventDriven <em>Event Driven</em>}</li>
 * </ul>
 * </p>
 *
 * @see de.uni_kassel.vs.cn.planDesigner.alica.AlicaPackage#getBehaviourConfiguration()
 * @model
 * @generated
 */
public interface BehaviourConfiguration extends AbstractPlan {
	/**
	 * Returns the value of the '<em><b>Parameters</b></em>' map.
	 * The key is of type {@link java.lang.String},
	 * and the value is of type {@link java.lang.String},
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Parameters</em>' map isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Parameters</em>' map.
	 * @see de.uni_kassel.vs.cn.planDesigner.alica.AlicaPackage#getBehaviourConfiguration_Parameters()
	 * @model mapType="de.uni_kassel.vs.cn.planDesigner.alica.EStringToEStringMapEntry<org.eclipse.emf.ecore.EString, org.eclipse.emf.ecore.EString>"
	 * @generated
	 */
	EMap<String, String> getParameters();

	/**
	 * Returns the value of the '<em><b>Deferring</b></em>' attribute.
	 * The default value is <code>"0"</code>.
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Deferring</em>' attribute isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Deferring</em>' attribute.
	 * @see #setDeferring(int)
	 * @see de.uni_kassel.vs.cn.planDesigner.alica.AlicaPackage#getBehaviourConfiguration_Deferring()
	 * @model default="0"
	 * @generated
	 */
	int getDeferring();

	/**
	 * Sets the value of the '{@link de.uni_kassel.vs.cn.planDesigner.alica.BehaviourConfiguration#getDeferring <em>Deferring</em>}' attribute.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Deferring</em>' attribute.
	 * @see #getDeferring()
	 * @generated
	 */
	void setDeferring(int value);

	/**
	 * Returns the value of the '<em><b>Frequency</b></em>' attribute.
	 * The default value is <code>"30"</code>.
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Frequency</em>' attribute isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Frequency</em>' attribute.
	 * @see #setFrequency(int)
	 * @see de.uni_kassel.vs.cn.planDesigner.alica.AlicaPackage#getBehaviourConfiguration_Frequency()
	 * @model default="30"
	 * @generated
	 */
	int getFrequency();

	/**
	 * Sets the value of the '{@link de.uni_kassel.vs.cn.planDesigner.alica.BehaviourConfiguration#getFrequency <em>Frequency</em>}' attribute.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Frequency</em>' attribute.
	 * @see #getFrequency()
	 * @generated
	 */
	void setFrequency(int value);

	/**
	 * Returns the value of the '<em><b>Behaviour</b></em>' container reference.
	 * It is bidirectional and its opposite is '{@link de.uni_kassel.vs.cn.planDesigner.alica.Behaviour#getConfigurations <em>Configurations</em>}'.
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Behaviour</em>' container reference isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Behaviour</em>' container reference.
	 * @see #setBehaviour(Behaviour)
	 * @see de.uni_kassel.vs.cn.planDesigner.alica.AlicaPackage#getBehaviourConfiguration_Behaviour()
	 * @see de.uni_kassel.vs.cn.planDesigner.alica.Behaviour#getConfigurations
	 * @model opposite="configurations" required="true" transient="false"
	 * @generated
	 */
	Behaviour getBehaviour();

	/**
	 * Sets the value of the '{@link de.uni_kassel.vs.cn.planDesigner.alica.BehaviourConfiguration#getBehaviour <em>Behaviour</em>}' container reference.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Behaviour</em>' container reference.
	 * @see #getBehaviour()
	 * @generated
	 */
	void setBehaviour(Behaviour value);

	/**
	 * Returns the value of the '<em><b>Event Driven</b></em>' attribute.
	 * The default value is <code>"false"</code>.
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Event Driven</em>' attribute isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Event Driven</em>' attribute.
	 * @see #setEventDriven(boolean)
	 * @see de.uni_kassel.vs.cn.planDesigner.alica.AlicaPackage#getBehaviourConfiguration_EventDriven()
	 * @model default="false"
	 * @generated
	 */
	boolean isEventDriven();

	/**
	 * Sets the value of the '{@link de.uni_kassel.vs.cn.planDesigner.alica.BehaviourConfiguration#isEventDriven <em>Event Driven</em>}' attribute.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Event Driven</em>' attribute.
	 * @see #isEventDriven()
	 * @generated
	 */
	void setEventDriven(boolean value);

} // BehaviourConfiguration
